src/engine/plugins/tile.cpp

```diff
@@ -205,10 +298,79 @@ Status TilePlugin::HandleRequest(const api::TileParameters &parameters, std::str
+                auto sum_node_weight =
+                    std::accumulate(forward_weight_vector.begin(), forward_weight_vector.end(), 0);
+
+                const auto coord_a = facade.GetCoordinateOfNode(
+                    forward_node_vector.size() > 1
+                        ? forward_node_vector[forward_node_vector.size() - 2]
+                        : edge.u);
+                const auto coord_b = facade.GetCoordinateOfNode(edge.v);
+
+                std::unordered_map<NodeID, int> c_nodes;
+
+                // Get all outgoing shortcuts
+                for (const auto adj_shortcut :
+                     facade.GetAdjacentEdgeRange(edge.forward_segment_id.id))
+                {
+                    std::vector<contractor::QueryEdge::EdgeData> unpacked_shortcut;
+
+                    if (!facade.GetEdgeData(adj_shortcut).forward)
+                    {
+                        continue;
+                    }
+
+                    routing_base.UnpackEdgeToEdges(edge.forward_segment_id.id,
+                                                   facade.GetTarget(adj_shortcut),
+                                                   unpacked_shortcut);
+
+                    if (unpacked_shortcut.size() < 2)
+                    {
+                        continue;
+                    }
+
+                    const auto first_geometry_id =
+                        facade.GetGeometryIndexForEdgeID(unpacked_shortcut[1].id);
+                    std::vector<NodeID> first_geometry_vector;
+                    facade.GetUncompressedGeometry(first_geometry_id, first_geometry_vector);
+
+                    auto sum_edge_weight = unpacked_shortcut[0].distance;
+
+                    auto turn_weight = sum_edge_weight - sum_node_weight;
+
+                    c_nodes.emplace(first_geometry_vector.front(), turn_weight);
+                }
+
+                const uint64_t angle_in =
+                    static_cast<uint64_t>(util::coordinate_calculation::bearing(coord_a, coord_b));
+
+                // Only write for those that have angles out
+                if (c_nodes.size() > 0)
+                {
+                    const auto angle_in_offset = use_point_value(angle_in);
+
+                    for (const auto possible_next_node : c_nodes)
+                    {
+                        const auto coord_c = facade.GetCoordinateOfNode(possible_next_node.first);
+                        const auto c_bearing = static_cast<uint64_t>(
+                            util::coordinate_calculation::bearing(coord_b, coord_c));
+
+                        const auto angle_out_offset = use_point_value(c_bearing);
+                        const auto angle_weight_offset = use_point_value(possible_next_node.second);
+
+                        // TODO this is not as efficient as it could be because of repeated
+                        // angles_in
+                        edge_turn_data.emplace_back(detail::TurnData{
```

Comment:
Good point.  Ideally, each value would only show up once in the layer attribute table.  But let's just get this shipped and make a ticket to remove duplicated values from the attribute table.